### PR TITLE
build avocado in venv support

### PR DESCRIPTION
**Pull request by suqinhuang**
need to test in venv if the machine is shared to several testers.
add the support to build avocado in venv

### CODE REVIEW

File: avocado/core/data_dir.py [hunk 1]
```diff
@@ -47,9 +47,27 @@
...
+
+def virtualenv_based_dir(path):
+    """
+    Return virtualenv based dir
+    """
+    virtualenv_path = os.environ['VIRTUAL_ENV']
+    path = os.path.join(virtualenv_path, path.lstrip(os.sep))
+    try:
+        os.makedirs(path)
+    except Exception:
```
Comment (CongSmile) [inline avocado/core/data_dir.py]: I think OSError is better.

**Comment by clebergnu** [comment]
@suqinhuang looks like with this change, while running the CI tests, is leaving some directories behind.

**Comment by suqinhuang** [comment]
@clebergnu 
1. we can test as normal without venv
2. the result of CI testing is save in xml file, and then read and show with plugin, so it should not effect the CI testing.

Regards

**Comment by clebergnu** [comment]
@suqinhuang do you mean you want to abandon this proposal? If so, please close this PR.

**Comment by suqinhuang** [comment]
@clebergnu hi, I mean this patch doesn't effect CI testing

**Comment by clebergnu** [comment]
@suqinhuang the point is that this patch did not pass the CI tests:

```
======================================================================
ERROR: test_replay_skip_skipped (selftests.functional.test_basic.RunnerHumanOutputTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/avocado-framework/avocado/selftests/functional/test_basic.py", line 377, in test_replay_skip_skipped
    "--replay-test-status PASS" % jobid))
  File "/home/travis/build/avocado-framework/avocado/avocado/utils/process.py", line 1027, in run
    raise CmdError(cmd, sp.result)
CmdError: Command './scripts/avocado run --replay 1ed871f18a13b9ab96aed3334ca944789fb2b944 --replay-test-status PASS' failed (rc=2)
======================================================================
FAIL: testDataDirFromConfig (selftests.unit.test_datadir.DataDirTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/avocado-framework/avocado/selftests/unit/test_datadir.py", line 48, in testDataDirFromConfig
    self.assertEqual(data_dir_func(), stg.get_value('datadir.paths', key))
AssertionError: '/home/travis/virtualenv/python2.7_with_system_site_packages/tmp/avocado_selftests.unit.test_datadirFsdmh9/tests' != '/tmp/avocado_selftests.unit.test_datadirFsdmh9/tests'
----------------------------------------------------------------------
Ran 288 tests in 75.725s
FAILED (failures=1, errors=1)
```

Please check the full output at: https://travis-ci.org/avocado-framework/avocado/jobs/109543033

**Comment by apahim** [comment]
@suqinhuang, just a minor comment. Please rebase and address @clebergnu comment before [v2].

File: avocado/core/data_dir.py (hunk 1 above)
Comment (apahim) [inline avocado/core/data_dir.py]: agreed.

**Comment by apahim** [comment]
Hi @suqinhuang, the end of the sprint in around the corner. Please update this PR if you want to give it a change to get in.

**Comment by suqinhuang** [comment]
hi @clebergnu @apahim 
too many error for ./selftests/run test, any idea to fix them, or just debug it one by one.

http://pastebin.test.redhat.com/353570

**Comment by apahim** [comment]
@suqinhuang the only error I face in my env is the first:
~~~
======================================================================
ERROR: testDataDirFromConfig (selftests.unit.test_datadir.DataDirTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/apahim/src/avocado/selftests/unit/test_datadir.py", line 51, in testDataDirFromConfig
    stg.get_value('datadir.paths', key)))
  File "/home/apahim/src/avocado/avocado/core/data_dir.py", line 55, in venv_based_dir
    venv_root = os.environ['VIRTUAL_ENV']
  File "/usr/lib64/python2.7/UserDict.py", line 23, in __getitem__
    raise KeyError(key)
KeyError: 'VIRTUAL_ENV'
~~~

This is caused by the line `55` in `data_dir.py`:
~~~
+    venv_root = os.environ['VIRTUAL_ENV']
~~~
And it could be avoided by using `venv_root = os.environ.get('VIRTUAL_ENV')`, which returns `None` when the variable is missing.

**Pull request closed without merging**